code/modules/projectiles/guns/energy.dm

```diff
@@ -198,7 +198,8 @@
 		if(R && R.cell)
 			var/obj/item/ammo_casing/energy/shot = ammo_type[select] //Necessary to find cost of shot
 			if(R.cell.use(shot.e_cost)) 		//Take power from the borg...
-				power_supply.give(shot.e_cost)	//... to recharge the shot
+				if(!istype(src, /obj/item/gun/energy/kinetic_accelerator/cyborg)) // Mining borg KA's can be spammed with no cooldown without this check

```

Comment:
This is such a ridiculous and egregious violation of OOP.

this needs to be addressed in a more fundamental way. This is absolutely not the way to do it.

Reply:
I know this is a bad fix and I don't know *how* to fix it other than doing some drastic change. Since KAs are subtypes of energy weapons, it passes through the `process_chamber()` proc on line 106 of the same energy.dm file and the `robocharge()` proc is called inside here.

As far as I can tell, you either make a check before calling `robocharge()` or you make a check inside of it, like I did. I don't see another way of doing this unless you made the borg KA not a subtype of energy weapon or something like that.